receiver/datadogreceiver/receiver.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadogreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/datadogreceiver"
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	datadogpb "github.com/DataDog/datadog-agent/pkg/trace/exportable/pb"
+	"github.com/gorilla/mux"
```

Comment:
One thing to be conscious of is that this project is no longer being maintained, so in a future PR we may need to change it.

Reply:
Tracked here: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18191

Reply:
I removed the gorilla mux dep